### PR TITLE
reference/tidb-binlog: add encrypt_password for tidb-binlog/deploy.md

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? 

Add encrypt_password for tidb-binlog/deploy.md

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [x] This PR is to align with: [pingcap/docs-cn#2185](https://github.com/pingcap/docs-cn/pull/2185)
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/tidb-binlog/deploy.md [hunk 1]
```diff
@@ -569,6 +569,9 @@ The following part shows how to use Pump and Drainer based on the nodes above.
         host = "192.168.0.13"
         user = "root"
         password = ""
+        # When encrypted_password is not empty, password encrypted using
+        # `./binlogctl -cmd encrypt -text string` will be ignored
```
Comment (lilin90) [inline dev/reference/tidb-binlog/deploy.md]: Please confirm with @july2993 whether the Chinese version mean:

**`encrypted_password` is encrypted using `./binlogctl -cmd encrypt -text string`.**
**When `encrypted_password` is not empty, the `password` above will be ignored.**

Comment (lilin90) [reply dev/reference/tidb-binlog/deploy.md]: And update all other descriptions in this PR if necessary. Thanks!

**Comment by gregwebs** [comment]
This process is very awkward. It would be better just to accept the password itself as an environment variable (the security is the same) without this encryption indirection and release that as a new version.

**Comment by ran-huang** [comment]
@lilin90 Thanks a lot! 
Already confirmed with @july2993 and applied all the suggestions. PTAL.

**Comment by lilin90** [comment]
> This process is very awkward. It would be better just to accept the password itself as an environment variable (the security is the same) without this encryption indirection and release that as a new version.

@july2993 What do you think?

**Pull request merged**